rest-api/src/main/java/uk/gov/gchq/gaffer/rest/application/ApplicationResourceConfig.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.rest.application;
+
+import io.swagger.jaxrs.config.BeanConfig;
+import io.swagger.jaxrs.listing.ApiListingResource;
+import io.swagger.jaxrs.listing.SwaggerSerializers;
+import org.glassfish.jersey.server.ResourceConfig;
+import uk.gov.gchq.gaffer.rest.SystemProperty;
+import uk.gov.gchq.gaffer.rest.serialisation.JacksonJsonProvider;
+import uk.gov.gchq.gaffer.rest.service.SimpleExamplesService;
+import uk.gov.gchq.gaffer.rest.service.SimpleGraphConfigurationService;
+import uk.gov.gchq.gaffer.rest.service.SimpleOperationService;
+import uk.gov.gchq.gaffer.rest.service.StatusService;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * An <code>ApplicationConfig</code> sets up the application resources and singletons.
+ */
+public class ApplicationResourceConfig extends ResourceConfig {
+    protected final Set<Class<?>> resources = new HashSet<>();
+
+    public ApplicationResourceConfig() {
+//        addSystemResources();
+//        addServices();
```

Comment:
Being able to register specific resources is important so any implementation/extensions can remove/override particular resources. Can we add the 'addServices' 'addSystemResources' methods back in?